Dawn/vale config

dawnkelly09 opened the pull request:
This PR:
- adds files for Vale configuration 
- adds files for GH Actions workflow related to Vale

Usage: Writers simply need to ensure they are using the updated main branch and then continue working with their documents as usual. 

Scroll through the conversation to see examples of flagged errors.

nhussein11 commented on .github/workflows/vale.yml:
```diff
@@ -0,0 +1,36 @@
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        id: get-modified-files
+        run: |
+          changed_files=$(gh pr diff --name-only ${{ github.event.pull_request.number }} | grep '\.md$' || true)
```
@0xLucca, just reviewing this line, maybe worth adding support to `.pages` files to avoid typos there as well:
```bash
changed_files=$(gh pr diff --name-only ${{ github.event.pull_request.number }} | grep '\.md$|.pages$' || true)
```

WDYT?

eshaben replied:
Yeah would also like it to catch capitalization of page titles if possible

eshaben commented on .github/styles/Papermoon/WordSwapList.yml:
```diff
@@ -0,0 +1,82 @@
+action:
+  name: replace
+swap:
+  "(?:API Console|dev|developer) key": API key
```
Can you please provide an example of what this does?

dawnkelly09 replied:
This swap list is a customization of the Google package one. I haven't really taken things out of it so there are likely things we don't want. 

This one will create a warning for "Use this instead of that". In the API instance, there is scripting so it will catch the different variations (API Console, API dev, etc) and generate a warning to use API key instead. We can safely get rid of anything in this list we don't feel is relevant.

eshaben replied:
So this overlaps with configs already provided by Google? I'd say we should clean this up to only relevant stuff that we want. Checked quickly and these are probably the only relevant ones, so I'd say just keep these unless they overlap with Google configs

"(?:file ?path|path ?name)": path
file name: filename
HTTPs: HTTPS
url: URL
wasm: Wasm

eshaben replied:
@dawnkelly09 this may have gotten overlooked

dawnkelly09 replied:
I kept a few others just for little language consistency things. This is the new list I'm proposing. Please let me know if you see anything else that you want removed. Thank you! 
swap:
  "(?:e-mail|Email|E-mail)": email
  "(?:file ?path|path ?name)": path
  "(?:kill|terminate|abort)": stop|exit|cancel|end
  "(?:ok|Okay)": OK|okay
  "(?:WiFi|wifi)": Wi-Fi
  '3\-D': 3D
  "un(?:check|select)": clear
  above: preceding
  admin: administrator
  Ajax: AJAX
  a\.k\.a|aka: or|also known as
  approx\.: approximately
  check box: checkbox
  disabled?: turn off|off|deactivate
  file name: filename
  HTTPs: HTTPS
  in order to: to
  open-source: open source
  stylesheet: style sheet
  tablename: table name
  url: URL
  vs\.: versus
  wasm: Wasm

eshaben replied:
sounds good thank you

eshaben commented on .github/styles/Papermoon/WordSwapList.yml:
```diff
@@ -0,0 +1,82 @@
+  file name: filename
+  firewalls: firewall rules
+  functionality: capability|feature
+  Google account: Google Account
+  Google accounts: Google Accounts
```
I would say Account probably doesn't need to be capitalized -- where did these all come from? haha some of them are weirdly specific

dawnkelly09 replied:
Read my previous comment. Google default stuff ;-)

eshaben commented on .github/styles/Papermoon/WordSwapList.yml:
```diff
@@ -0,0 +1,82 @@
+  in order to: to
+  ingest: import|load
+  k8s: Kubernetes
+  long press: touch & hold
```
avoid ampersands

dawnkelly09 replied:
Fixed in new commit

eshaben commented on .github/styles/Papermoon/WordSwapList.yml:
```diff
@@ -0,0 +1,82 @@
+  open-source: open source
+  overview screen: recents screen
+  regex: regular expression
+  SHA1: SHA-1|HAS-SHA1
```
What about this one?

dawnkelly09 replied:
I'll remove that, too.

eshaben commented on .github/styles/config/vocabularies/Polkadot/accept.txt:
```diff
@@ -0,0 +1,26 @@
+parathread
+Polkadot
+Polkadot's
+relaychain
+relaychains
```
relay chain is two words

dawnkelly09 replied:
Fixed in new commit. thanks

eshaben commented on .github/styles/config/vocabularies/Polkadot/accept.txt:
```diff
@@ -0,0 +1,26 @@
+bootnodes
+config
+[Cc]oretime
+dev_newBlock
+dev_setHead
+dev_setStorage
+dev_timeTravel
```
were you able to figure out a way for ignoring the stuff in quotes, like `??? function "THIS STUFF"`

dawnkelly09 replied:
Not yet. I got it to ignore the params in those instances but still working on the function names. Haven't landed on the regex pattern yet that won't also grab other stuff.

dawnkelly09 replied:
Once I have the vocab beefed up (need to screen more files) and get the existing rules in place, I will start making a list of things from the style guide still needing enforced and open some tickets so we can track them as they are added to the config, etc.

eshaben commented on docs/dev-tools/chopsticks/overview.md:
```diff
@@ -18,7 +18,7 @@ With Chopsticks, developers can simulate and test complex blockchain scenarios w
 
 !!! note
-    Chopsticks uses [Smoldot](https://github.com/smol-dot/smoldot){target=_blank} light client, which only supports the native Polkadot-SDK API. Consequently, a Chopsticks-based fork does not support Ethereum JSON-RPC calls, so you cannot use it to fork your chain and connect Metamask.
+    Chopsticks uses [Smoldot](https://github.com/smol-dot/smoldot){target=_blank} light client, which only supports the native Polkadot-SDK API. Consequently, a Chopsticks-based fork doesn't support Ethereum JSON-RPC calls, so you cannot use it to fork your chain and connect Metamask.
```
Just happened to notice this, but Polkadot SDK doesn't need a dash between it

dawnkelly09 replied:
Fixed. Good eye. Updated the vocab to catch this, too.

eshaben reviewed (commented):
Great job! Super exciting! Had some questions

dawnkelly09 commented on .github/styles/Papermoon/Titlecase.yml:
```diff
@@ -0,0 +1,14 @@
+
+exceptions:
+  - ABC
+  - add
```
Nice catch! This was a holdover from the Google package rule. They use sentence case for headings and must have a specific need to ignore add in that situation. I will remove it.

dawnkelly09 commented on .github/styles/Papermoon/Titlecase.yml:
```diff
@@ -0,0 +1,14 @@
+style: Chicago
+
+exceptions:
+  - ABC
```
This one we actually do want! It allows headings which start with consecutive capital letters to pass the check. This prevents flagging instances where the heading starts with an acronym

dawnkelly09 commented:
@eshaben I think I addressed all the comments, etc. Please let me know if you have more thoughts or requests for Vale setup. I am collecting a list of things we want it to do that aren't in place yet.

eshaben reviewed (approved):
LGTM! On the condition we circle back and address this comment in a future PR: https://github.com/papermoonio/polkadot-ecosystem-docs-draft/pull/32#discussion_r1727681076
